pkg/engine/armvariables.go

```diff
@@ -147,6 +147,60 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 		"vnetResourceGroupNameResourceSegmentIndex": 4,
 	}
 
+	if cs.Properties.IsUbuntuDistroForAllNodes() {
+		masterVars["cloudInitFiles"] = map[string]interface{}{
```

Comment:
slight preference for doing something like:

```
masterVars["cloudInitFiles"] = map[string]interface{}{
			"provisionScript":                  getBase64EncodedGzippedCustomScript(kubernetesCSEMainScript),
			"provisionSource":                  getBase64EncodedGzippedCustomScript(kubernetesCSEHelpersScript),
			"provisionInstalls":                getBase64EncodedGzippedCustomScript(kubernetesCSEInstall),
			"provisionConfigs":                 getBase64EncodedGzippedCustomScript(kubernetesCSEConfig),
			"provisionCIS":                     getBase64EncodedGzippedCustomScript(kubernetesCISScript),
			"healthMonitorScript":              getBase64EncodedGzippedCustomScript(kubernetesHealthMonitorScript),
			"customSearchDomainsScript":        getBase64EncodedGzippedCustomScript(kubernetesCustomSearchDomainsScript),
			"generateProxyCertsScript":         getBase64EncodedGzippedCustomScript(kubernetesMasterGenerateProxyCertsScript),
			"mountEtcdScript":                  getBase64EncodedGzippedCustomScript(kubernetesMountEtcd),
			"kubeletSystemdService":            getBase64EncodedGzippedCustomScript(kubeletSystemdService),
			"kmsSystemdService":                getBase64EncodedGzippedCustomScript(kmsSystemdService),
			"kubeletMonitorSystemdTimer":       getBase64EncodedGzippedCustomScript(kubernetesKubeletMonitorSystemdTimer),
			"kubeletMonitorSystemdService":     getBase64EncodedGzippedCustomScript(kubernetesKubeletMonitorSystemdService),
			"dockerMonitorSystemdTimer":        getBase64EncodedGzippedCustomScript(kubernetesDockerMonitorSystemdTimer),
			"dockerMonitorSystemdService":      getBase64EncodedGzippedCustomScript(kubernetesDockerMonitorSystemdService),
			"aptPreferences":                   getBase64EncodedGzippedCustomScript(aptPreferences),
			"dockerClearMountPropagationFlags": getBase64EncodedGzippedCustomScript(dockerClearMountPropagationFlags),
			"etcdSystemdService":               getBase64EncodedGzippedCustomScript(etcdSystemdService),
		}

if cs.Properties.IsUbuntuDistroForAllNodes() {
     // append all the ubuntu distro specific vars
}

```

That way there are no duplicated lines and we avoid risking someone making a mistake and only adding a new var for one of the two cases.

Reply:
<img width="1493" alt="image" src="https://user-images.githubusercontent.com/1494193/56608000-b10b2500-65be-11e9-9b31-d52ece6dda6c.png">

Reply:
`masterVars["cloudInitFiles"].(map[string]interface{})["etcIssue"] = getBase64EncodedGzippedCustomScript(etcIssue)` should work